packit_service/worker/handlers/open_scan_hub.py

```diff
@@ -0,0 +1,125 @@
+    ConfigFromEventMixin,
+)
+from packit_service.worker.result import TaskResults
+from packit_service.worker.mixin import (
+    LocalProjectMixin,
+    PackitAPIWithUpstreamMixin,
+)
+
+from packit_service.worker.reporting import BaseCommitStatus
+
+logger = logging.getLogger(__name__)
+
+
+@configured_as(job_type=JobType.copr_build)
+@reacts_to(OpenScanHubTaskFinishEvent)
+class OpenScanHubTaskFinishHandler(
+    RetriableJobHandler,
+    LocalProjectMixin,
+    ConfigFromEventMixin,
+    PackitAPIWithUpstreamMixin,
+):
+    task_name = TaskName.openscanhub_task_finish
+
+    @staticmethod
+    def get_checkers() -> Tuple[Type[Checker], ...]:
+        return ()
+
+    def run(self) -> TaskResults:
+        task_id = self.data.event_dict["task_id"]
+        event = self.data.to_event()
+        if not event.scan or not event.build:
+            return TaskResults(
+                success=False,
+                details={
+                    "msg": f"Scan {task_id} not found or not associated with a Copr build"
+                },
+            )
+        elif not self.job_config:
+            return TaskResults(
+                success=False,
+                details={
+                    "msg": (
+                        "No job configuration found for "
+                        f"openscanhub_task_finish in {self.project.repo}"
+                    )
+                },
+            )
+
+        branches = aliases.get_build_targets(
+            *self.job_config.targets,
+        )
+        if "fedora-rawhide-x86_64" not in branches:
+            return TaskResults(
+                success=False,
+                details={
+                    "msg": "Skipping job configuration with no fedora-rawhide-x86_64 target."
+                },
+            )
+
+        build_helper = CoprBuildJobHelper(
+            service_config=self.service_config,
+            package_config=self.package_config,
+            project=self.project,
+            metadata=self.data,
+            db_project_event=self.data.db_project_event,
+            job_config=self.job_config,
+            celery_task=self.celery_task,
+        )
+
+        scan_helper = ScanHelper(copr_build_helper=build_helper, build=event.build)
+
+        external_links = {
+            "Added issues": event.issues_added_url,
+            "Fixed issues": event.issues_fixed_url,
+            "Scan results": event.scan_results_url,
+        }
+
+        # TODO: probably we need a babysit task for when the build is not finished yet
+        if event.build.status == "success":
+            state = BaseCommitStatus.success
+            description = (
+                "Scan in OpenScanHub is finished. Check the URL for more details."
+            )
+        else:
+            state = BaseCommitStatus.neutral
+            description = (
+                "Scan in OpenScanHub is finished but the build did not "
+                "finish yet or did not succeed."
```

Comment:
We're talking about RPM build in Copr, right? Cause from the wording I'm a bit confused :D SRPM build triggers OSH. Now that I think about it… Does it matter whether the RPM build succeeds or fails? :thinking: OSH does static analysis, the results should be relevant regardless

Reply:
we trigger OSH just when the RPM build finishes successfully, so that we do not overload OSH. But there is mock build being done again after the submission in OSH and scan is being run when the mock build finishes successfully, am I right @siteshwar ?

Reply:
> we trigger OSH just when the RPM build finishes successfully, so that we do not overload OSH. But there is mock build being done again after the submission in OSH and scan is being run when the mock build finishes successfully, am I right @siteshwar ?

OpenScanHub uses a wrapper over `mock` which is known as `csmock` that is used to capture and process the output of a RPM build, so basically you rebuild the entire package again. As an example, you can see the [stdout.log](https://openscanhub.fedoraproject.org/task/15319/log/stdout.log) for the curl package.

Reply:
Sorry for the noise, and yes I was talking about the copr build. I didn't know where to take the result state for the scan and I thought that as a **first step** implementation we wanted just to mark the check as finished and mirroring the state of the "copr build"...
Now that I know where to take the info I will handle the state of the scan and change the check description!